qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -825,13 +935,24 @@ def _run_analysis(
         if isinstance(fit_options, FitOptions):
             fit_options = [fit_options]
 
+        # Prepare fit model
+        prepared_fit_func = self.fit_model
+
+        if self.__fixed_parameters__:
+            fixed_params = {p: self.options.get(p) for p in self.__fixed_parameters__}
+            prepared_fit_func.bind_parameters(**fixed_params)
+        else:
+            fixed_params = None
```

Comment:
If we move the fixed parameters to default options and initialize the `fit_model` with them we can get ride of this logic.